omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPersonResourceProvider.java

```diff
@@ -120,4 +124,26 @@ public MethodOutcome createFHIRPerson(@ResourceParam Person person) {
 		return retVal;
 	}
 
+	@Update
+	public MethodOutcome updatePatientConditional(
+	      @ResourceParam Person thePerson,
+	      @IdParam IdDt theId) {
+	   
+	   MethodOutcome retVal = new MethodOutcome();
+	   retVal.setId(new IdDt(FHIRConstants.PERSON, thePerson.getId().getIdPart()));		    
+	   OperationOutcome outcome = new OperationOutcome();
+	   
+	   try{
+		   Person person=personResource.updateFHIRPerson(thePerson,theId.getIdPart());
```

Comment:
I'll corect the spaces. "theId" never becomes empty. It is comming from the uri. but "thePerson.getId().getIdPart()" can be empty, I'll handle that.